src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -532,42 +526,21 @@ public override void SetAsyncCallbacks(SNIAsyncCallback receiveCallback, SNIAsyn
         /// <returns>SNI error code</returns>
         public override uint SendAsync(SNIPacket packet, SNIAsyncCallback callback = null)
         {
-            SNIPacket newPacket = packet;
-
-            _writeTaskFactory.StartNew(() =>
-            {
-                try
-                {
-                    lock (this)
-                    {
-                        packet.WriteToStream(_stream);
-                    }
-                }
-                catch (Exception e)
-                {
-                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.TCP_PROV, SNICommon.InternalExceptionError, e);
-
-                    if (callback != null)
-                    {
-                        callback(packet, TdsEnums.SNI_ERROR);
-                    }
-                    else
+            Task writeTask = null;
+            writeTask = packet.WriteToStreamAsync(_stream);
+            writeTask.ConfigureAwait(false);
```

Comment:
This is a nop.  ConfigureAwait is a helper that returns a struct that influences how await operates; it doesn't mutate the task on which it's called at all.  You can simply delete this line.

That said, related to my previous comment about overheads in the reading method, even more than reads writes often complete synchronously.  It'd be nice to look at removing the additional overheads here as well.